File: python/atelier_app/seed.py
"""Console output helpers for the `seed` command."""

from __future__ import annotations

import atelier_bench.rules as rules
from atelier_bench import METRICS

from .state import AppState


def summarize(state: AppState) -> str:
    data = state.data
    return ", ".join(
        [
            f"seeded: {len(data.customers)} customer(s)",
            f"{len(data.orders)} order(s)",
            f"{len(data.parts)} part(s)",
            f"revenue {data.revenue_cents()}c",
        ]
    )


def metric_lines(state: AppState) -> list[str]:
    return [f"{entry.key} = {entry.formatted(state.data)}" for entry in METRICS]


def rule_line(state: AppState) -> str:
    return f"rules: {len(rules.RuleRegistry.satisfied(state.data))}/{len(rules.RULES)} satisfied"

File: go/app/seed.go
package app

import (
	"fmt"

	"atelier.example/lane/atelier/metrics"
	"atelier.example/lane/atelier/rules"
)

// Summarize is the seed line every lane prints.
func Summarize(state *State) string {
	data := state.Data
	return fmt.Sprintf(
		"seeded: %d customer(s), %d order(s), %d part(s), revenue %dc",
		len(data.Customers),
		len(data.Orders),
		len(data.Parts),
		data.RevenueCents(),
	)
}

// MetricLines renders every registered metric, in sorted key order.
func MetricLines(state *State) []string {
	lines := make([]string, 0, metrics.Count())
	for _, metric := range metrics.All() {
		lines = append(lines, fmt.Sprintf("%s = %.2f %s", metric.Key(), metric.Compute(state.Data), metric.Unit()))
	}
	return lines
}

// RuleLine renders the satisfied-rule tally.
func RuleLine(state *State) string {
	return fmt.Sprintf("rules: %d/%d satisfied", len(rules.Satisfied(state.Data)), rules.Count())
}
